modules/adblocker/sources/adblocker.es

```diff
@@ -116,6 +153,53 @@ export default class Adblocker {
     this.whitelistChecks.push(fn);
   }
 
+  /**
+   * Given a webNavigation.onCommitted event, decide whether the event should
+   * be handled. As a rule of thumb, all events should be forwarded to the
+   * adblocker engine unless there an exception has been configured in Ghostery.
+   *
+   * For instance, if the site has is been visited has been trusted by the user,
+   * the event must not be forwarded.
+   *
+   * Input: "details" has the same structure as an event
+   * from "webNavigation.onCommitted".
+   */
+  shouldProcessOnCommittedEvent(details) {
+    const skip = (reason) => {
+      logger.log('Ignore visit to', details.url, ':', reason);
+      return false;
+    };
+
+    if (this.isReady() === false) {
+      return skip('adblocker engine is not ready');
+    }
+    if (details.url === 'about:blank') {
+      // TODO: Should we skip it?
+      return true;
+    }
```

Comment:
I think we should potentially forward these? (e.g. when there is an anonymous iframe without a source). In any case it should not cause any issue in the adblocker.

Reply:
That was also what I thought. It is not directly the same as the matchAboutBlank topic, since the important part is to see the first navigation to the page and then inject to all frames. But then there are still onCommitted events being sent, which I don't exactly understand what they mean. On the other hand, I also think the adblocker knows how to deal with it, so unless there is a compelling reason to drop them, we should forward them.

(Note: Unless there are indicators that forwarding leads to correctness problems, we could then drop the comment. Only filtering them for performance reasons should not be worth it. The reason why the check now exists is because the following tests rely on the URL having a hostname)